### PR TITLE
show project cache usage stats in widget

### PR DESCRIPTION
**Pull request by hjelmevold**
In order to help Booster users debug their cache configuration, the Booster widget in Content Studio now not only allows you to purge cache for the project, but it also shows cache usage statistics.

- Shows a table with up to 20 of the most cached paths that belong to the current project, sorted descending by number of cache entries.
- Each path has a details element that allows for drilling down into the cache for the current path, with how many cached variations there are for certain querystring parameters. This allows users to discover that an inconsequential parameter (e.g. `?sometrackingparam=abc123`) is filling up unnecessarily many cache entries when there is otherwise no change in rendered content between these variations.
- When each details element is expanded on mouse click, its details are fetched with AJAX, and only fetched upon first expansion attempt.

### CODE REVIEW

File: src/main/resources/admin/widgets/booster/booster.js [hunk 1]
```diff
@@ -28,6 +29,30 @@ const isAppEnabledOnSite = (contentId) => {
...
+const getCommonlyCachedPaths = (project, numResults) => {
+    const boosterRepo = nodeLib.connect({
+        repoId: 'com.enonic.app.booster',
+        branch: 'master'
+    });
+
+    const allCachedPaths = boosterRepo.query({
+        start: 0,
+        count: 0,
+        query: `project = '${project}'`,
```
Comment (rymsha) [inline src/main/resources/admin/widgets/booster/booster.js:41]: use query dsl

File: src/main/resources/services/pathstats/pathstats.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+function parseQueryString(queryString) {
+  if (!queryString) {
+    return null;
+  }
+
+  return queryString.split('&').reduce((acc, pair) => {
+    const parts = pair.split('=');
+    const key = parts[0];
+    const value = parts[1];
```
Comment (rymsha) [inline src/main/resources/services/pathstats/pathstats.js:13]: may fail for legal `?q`

File: src/main/resources/services/pathstats/pathstats.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+  const boosterRepo = nodeLib.connect({
+    repoId: 'com.enonic.app.booster',
+    branch: 'master'
+  });
+
+  const pathUrls = boosterRepo.query({
+    start: 0,
+    count: 0,
+    query: 'path = "' + path + '"',
```
Comment (rymsha) [inline src/main/resources/services/pathstats/pathstats.js:81]: use query dsl

File: src/main/resources/admin/widgets/booster/booster.js (hunk 1 above)
Comment (hjelmevold) [reply src/main/resources/admin/widgets/booster/booster.js]: Done. I pushed new commit 3b6af24671ef3a624765c47e8757f82c508a3b13

File: src/main/resources/services/pathstats/pathstats.js (hunk 3 above)
Comment (hjelmevold) [reply src/main/resources/services/pathstats/pathstats.js]: Done. I pushed new commit 3b6af24671ef3a624765c47e8757f82c508a3b13

File: src/main/resources/services/pathstats/pathstats.js (hunk 2 above)
Comment (hjelmevold) [reply src/main/resources/services/pathstats/pathstats.js:13]: I tested with a querystring like `?q&r=&s` and in that case all three params are counted, and I've also tested that the value is counted as `null` no matter if the param key is solo like `?q` or with empty value like `?q=`

There might be other edge cases that I don't know about, though.
